chore(s3): readme update with mixing L1 and L2 bucket policy (#31437)
aws/aws-cdk: packages/aws-cdk-lib/aws-s3/README.md

});
```

The above code will create a new bucket policy if none exists or update the

> existing bucket policy otherwise to allow access log delivery.

Reviewer: ```suggestion
existing bucket policy to allow access log delivery.
```

However, there could be an edge case if the `accessLogsBucket` also defines a bucket

> policy resource using the L1 Construct. Although the mix of L1 and L2 Constructs are not
> recommended, there is nothing stopping users to do this at the moment.

Reviewer: ```suggestion
policy resource using the L1 Construct. Although the mixing of L1 and L2 Constructs is not
recommended, there are no mechanisms in place to prevent users from doing this at the moment.
```

```ts

> const bucketName = "my-favorite-bucket-name";

Reviewer: <3

const accessLogsBucket = new s3.Bucket(this, 'AccessLogsBucket', {
  objectOwnership: s3.ObjectOwnership.BUCKET_OWNER_ENFORCED,
  bucketName,
});

// Creating a bucket policy using L1
const bucketPolicy = new s3.CfnBucketPolicy(this, "BucketPolicy", {
  bucket: bucketName,
  policyDocument: {
    Statement: [
      {
        Action: 's3:*',
        Effect: 'Deny',
        Principal: {
          AWS: '*',
        },
        Resource: [
          accessLogsBucket.bucketArn,
          `${accessLogsBucket.bucketArn}/*`
        ],
      },
    ],
    Version: '2012-10-17',
  },
});

// 'serverAccessLogsBucket' will create a new L2 bucket policy
// to allow log delivery and overwrite the L1 bucket policy.
const bucket = new s3.Bucket(this, 'MyBucket', {
  serverAccessLogsBucket: accessLogsBucket,
  serverAccessLogsPrefix: 'logs',
});
```

> The above example uses L2 Bucket Construct with L1 CfnBucketPolicy Construct. However,
> when `serverAccessLogsBucket` is used, it will attempt to create a new L2 Bucket Policy
> resource and overwrite the permissions defined in the L1 Bucket Policy, causing unintended
> behaviours.

Reviewer: ```suggestion
The above example uses the L2 Bucket Construct with the L1 CfnBucketPolicy Construct. However,
when `serverAccessLogsBucket` is set, a new L2 Bucket Policy
resource will be created which overwrites the permissions defined in the L1 Bucket Policy causing unintended
behaviours.
```
